### ISSUE TITLE
Notification left-click action

### ISSUE THREAD

**Issue by musjj** [awesomeWM/awesome#3850]
<details>
<summary>Output of `awesome --version`:</summary>

```
awesome 4.3 (Too long)
 • Compiled against Lua 5.1.4 (running with LuaJIT 2.1.0-beta3)
 • API level: 4
 • D-Bus support: yes
 • xcb-errors support: no
 • execinfo support: yes
 • xcb-randr version: 1.6
 • LGI version: module 'lgi.version' not found:
	no field package.preload['lgi.version']
	no file './share/lua/5.1/lgi/version.lua'
	no file './lgi/version.lua'
	no file './lgi/version/init.lua'
	no file './lib/lua/5.1/lgi/version.so'
	no file './lgi/version.so'
	no file './lib/lua/5.1/loadall.so'
	no file './lib/lua/5.1/lgi.so'
	no file './lgi.so'
	no file './lib/lua/5.1/loadall.so'
 • Transparency enabled: yes
 • Custom search paths: yes
```
</details>

**How to reproduce the issue:**

I'm not sure how to reproduce the issue minimally, but one tedious way to do it:

1. Turn on push notifications on a social media website (twitter, youtube, etc.)
2. Wait for a notification to come
3. Click on the notification

**Actual result:**

Clicking on the notification should open the associated post.

**Expected result:**

No action is performed.

With `dunst`, I have the following config:

```lua
mouse_left_click = do_action, close_current
```

Clicking on a notification window will perform an action. For example, clicking on a notification from twitter will open the associated post in the web browser. But I'm unable to recreate this same behavior with naughty notifications.

In dunst, there's this code:

https://github.com/dunst-project/dunst/blob/b032c7e15bb3357dc61fbe513bb3776b607cda24/src/input.c#L96-L106

https://github.com/dunst-project/dunst/blob/b032c7e15bb3357dc61fbe513bb3776b607cda24/src/notification.c#L705-L723

https://github.com/dunst-project/dunst/blob/b032c7e15bb3357dc61fbe513bb3776b607cda24/src/dbus.c#L890-L913

And it looks like there's similar code in awesomewm with:

https://github.com/awesomeWM/awesome/blob/aa8c7c6e27a20fa265d3f06c5dc3fe72cc5f021e/lib/naughty/dbus.lua#L159-L163

But it doesn't seem to be doing its thing. I'm really not familiar with the notification API, so some clarification would be really appreciated.

One thing I don't get is that web browser notifications don't display any "action" buttons that you'd see if you do something like `notify-send -A ...`. But I assume that there *is* an associated action somehow, since it works with `dunst`.

**Comment by actionless** [comment]
it must be broken somewhere during last 2-3 years - i remember it was broken long before (in a different manner) and then it got fixed, but since then i never tested that function again

**Comment by anivelle** [comment]
Hello, popping in because I've also noticed this issue.

@actionless did you mean that the `sendActionInvoked()` function was broken? It seems to me that `args.run()` is just never called according to the current `naughty.notification` implementation. If you meant the former, I'd be willing to test it out again sometime this weekend, if it hasn't been yet.

**Comment by actionless** [comment]
i didn't meant anything extra

according to blank log - yup, probably event not even firing or there is some if condition ni the callback which never True

**Comment by anivelle** [comment]
I think what happened is that when a notification is received from DBus, the default action is skipped (if that's the right word for it) so that it doesn't appear as an action button:
https://github.com/awesomeWM/awesome/blob/7ed4dd620bc73ba87a1f88e6f126aed348f94458/lib/naughty/dbus.lua#L162-L167

Then in the `naughty.notification` constructor `args.run()` is never referenced, nor is it referenced anywhere else except `naughty.layout.legacy`. Instead there is the `invoked_cb()` function in the constructor,  which appears to emit the `invoked` signal, while being attached to it:
https://github.com/awesomeWM/awesome/blob/7ed4dd620bc73ba87a1f88e6f126aed348f94458/lib/naughty/notification.lua#L1067-L1076 
https://github.com/awesomeWM/awesome/blob/7ed4dd620bc73ba87a1f88e6f126aed348f94458/lib/naughty/notification.lua#L828
I'm not exactly sure what is meant by resident notification, but I assume it's a persistent one. 

If that sounds about right to you would it be cool if I tried to fix it? I'm new to open source contribution but I'd like to help in any way I can.

**Comment by actionless** [comment]
yup would be really dope if you got time to look into it further and create a PR 😺👍

**Comment by anivelle** [comment]
Absolutely! I'm a little busy today but I should have more time tomorrow to get into it

**Comment by anivelle** [comment]
So, actually, doing some digging into things that aren't code, this issue has apparently already been solved with [this PR](https://github.com/awesomeWM/awesome/pull/3863) from October. I thought my Awesome was updated, but it looks like the AUR package doesn't get marked updated or something so I had to manually do it, and I can confirm this works. As is, like the author mentions it's a little rough, since clicking any other action will also send the default action, which is somewhat unintended. They mentioned they would try to get a proper fix as well but I haven't seen anything yet. Since that's the case I think I'll continue here to see if I can work on that solution.
